config/index.ts

```diff
@@ -1,268 +1,405 @@
-  if (useHiddenConfig || CLIConfiguration.isActive()) {
-    return getConfigFilePath();
-  }
-  return config_DEPRECATED.getConfigPath(path);
+export function createEmptyConfigFile(useGlobalConfig = false): void {
+  const { configFilePathFromEnvironment } = getConfigPathEnvironmentVariables();
+  const defaultPath = useGlobalConfig
+    ? getGlobalConfigFilePath()
+    : getLocalConfigFileDefaultPath();
+
+  const pathToWrite = configFilePathFromEnvironment || defaultPath;
+
+  writeConfigFile({ accounts: [] }, pathToWrite);
 }
 
-export function configFileExists(useHiddenConfig?: boolean) {
-  return useHiddenConfig
-    ? newConfigFileExists()
-    : Boolean(config_DEPRECATED.getConfigPath());
+export function deleteConfigFile(): void {
+  const pathToDelete = getConfigFilePath();
+  fs.unlinkSync(pathToDelete);
 }
 
-export function getAccountConfig(accountId?: number): CLIAccount | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getConfigForAccount(accountId);
+export function getConfigAccountById(accountId: number): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    'accountId',
+    accountId
+  );
+
+  if (!account) {
+    throw new Error(i18n('config.getConfigAccountById.error', { accountId }));
   }
-  return config_DEPRECATED.getAccountConfig(accountId) || null;
+
+  return account;
 }
 
-export function accountNameExistsInConfig(name: string): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.isAccountInConfig(name);
+export function getConfigAccountByName(
+  accountName: string
+): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(accounts, 'name', accountName);
+
+  if (!account) {
+    throw new Error(
+      i18n('config.getConfigAccountByName.error', { accountName })
+    );
   }
-  return config_DEPRECATED.accountNameExistsInConfig(name);
+
+  return account;
 }
 
-export function updateAccountConfig(
-  configOptions: Partial<FlatAccountFields>
-): FlatAccountFields | null {
-  const accountIdentifier = getAccountIdentifier(configOptions);
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.addOrUpdateAccount({
-      ...configOptions,
-      accountId: accountIdentifier,
-    });
+export function getConfigDefaultAccount(): HubSpotConfigAccount {
+  const { accounts, defaultAccount } = getConfig();
+
+  if (!defaultAccount) {
+    throw new Error(i18n('config.getConfigDefaultAccount.fieldMissingError'));
   }
-  return config_DEPRECATED.updateAccountConfig({
-    ...configOptions,
-    portalId: accountIdentifier,
-  });
-}
 
-export function updateDefaultAccount(nameOrId: string | number): void {
-  if (CLIConfiguration.isActive()) {
-    CLIConfiguration.updateDefaultAccount(nameOrId);
-  } else {
-    config_DEPRECATED.updateDefaultAccount(nameOrId);
+  const account = getConfigAccountByInferredIdentifier(
```

Comment:
This function allows you to get accounts by name or id (primarily for backwards compatibility with how the CLI currently works), but opted to use it as minimally as possible to reduce likelihood of bugs.

We now only _set_ default accounts using their `accountId`, but this is set up to maintain backwards compatibility with default accounts specified by `name`